src/Input/Select/Select.tsx

```diff
@@ -157,6 +158,18 @@ const Select: ISelect = (props: Props) => {
     [onFocus]
   );
 
+  const handleClick = () => {
+    setIsFocus(!isFocus);
+  };
+
+  const handleDropIconClick = () => {
+    setIsFocus(!isFocus);
+
+    if (!disableTyping && !isFocus) {
+      selectInputRef.current.focus();
+    }
```

Comment:
I am not sure whether `setIsFocus` can be run synchronously, so this is safer:
```
if (!disableTyping && !isFocus) {
      selectInputRef.current.focus();
}
setIsFocus(!isFocus);
```

Reply:
and why don't wrap the function in useCallback?

Reply:
Because it is passed to an `input` rather than `PureComponent`. 🤔

Reply:
yes, you are right.